apps/client/src/pages/QuestionsPage.js

```diff
@@ -113,11 +114,20 @@ const QuestionsPage = ({ topic, checker }) => {
         })
       );
     } else {
-      // Go back to the Location page
-      history.push(geturl(routes.address, { slug }));
+      goBack();
     }
   };
 
+  const goBack = () => {
+    const route = getDataNeedResultPageOrPrevious(
+      checker,
+      autofillRoutes,
+      routes
+    );
+    const url = geturl(route, topic);
+    history.push(url);
+  };
```

Comment:
This function declaration if placed below the call. Any reason?

Reply:
No reason. Do we not do this?

Reply:
fixed ;-p